edx_event_bus_kafka/internal/tests/test_consumer.py

```diff
@@ -305,6 +305,41 @@ def raise_exception():
         assert mock_emit.call_args_list == [call(self.normal_message)] * exception_count
         assert exc_info.value.args == ("Too many consecutive errors, exiting (4 in a row)",)
 
+    @override_settings(
+        EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL='http://localhost:12345',
+        EVENT_BUS_KAFKA_BOOTSTRAP_SERVERS='localhost:54321',
+        EVENT_BUS_TOPIC_PREFIX='dev',
+    )
+    def test_connection_reset_if_unusable(self):
+        """Confirm we reconnect to the database if our connection is deemed unusable"""
+        mock_emit = Mock(side_effect=side_effects([self.event_consumer._shut_down]))  # pylint: disable=protected-access
+        with patch.multiple(self.event_consumer,
+                            emit_signals_from_message=mock_emit,
+                            **{'_db_needs_reconnect': Mock(return_value=True), '_reconnect_to_db': DEFAULT}
+                            ) as mock_event_consumer:
+            mock_consumer = Mock(**{'poll.return_value': self.normal_message}, autospec=True)
+            self.event_consumer.consumer = mock_consumer
+            self.event_consumer.consume_indefinitely()
+        mock_event_consumer['_reconnect_to_db'].assert_called_once()
```

Comment:
1. I'm used to seeing something like: `mock_event_consumer._reconnect_to_db.assert_called_once()`. Does either work, or am I just not getting this code at all? I didn't even know you could get the method by key, but maybe I'm missing something and this is different from what I think.
2. What does `DEFAULT` do?

Reply:
[Fixed]  Refactored to not use `patch.multiple`.

[inform] getting the method by key is how  the return value of `patch.multiple()` works. DEFAULT just means use a default mock (no custom return method or anything).